components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -915,25 +913,31 @@ private static List<String> getOIDCScopeNames() throws IdentityOAuthAdminExcepti
         return Arrays.asList(ArrayUtils.nullToEmpty(oAuthAdminService.getScopeNames()));
     }
 
-    private static List<String> getAllowedOAuthScopes(OAuth2Parameters params) throws OAuthSystemException {
+    /**
+     * Drop OIDC and unregistered scopes from consent required scopes.
+     *
+     * @param params OAuth2 parameters.
+     * @return consent required scopes
+     * @throws OAuthSystemException If retrieving OIDC scopes failed.
+     */
+    private static List<String> dropOIDCAndUnregisteredScopesFromConsentRequiredScopes(OAuth2Parameters params)
+            throws OAuthSystemException {
 
         Set<String> allowedScopes = params.getScopes();
         List<String> allowedOAuthScopes = new ArrayList<>();
         if (CollectionUtils.isNotEmpty(allowedScopes)) {
             try {
                 startTenantFlow(params.getTenantDomain());
-
-            /* If DropUnregisteredScopes scopes config is enabled
-             then any unregistered scopes(excluding internal scopes
-             and allowed scopes) is be dropped. Therefore they will
-             not be shown in the user consent screen.*/
+                /* If DropUnregisteredScopes scopes config is enabled
+                 then any unregistered scopes(excluding internal scopes
```

Comment:
comment can be added through one to two lines (one line can be up to 120 length limit)